src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1192 @@ const ActorInit En_Niw_InitVars = {
+    if (this->actor.bgCheckFlags & 0x20) {
+        this->actor.gravity = 0.0f;
+
+        if (15.0f < this->actor.waterY) {
+            this->actor.posRot.pos.y += 2.0f;
+        }
+
+        if (this->timer4 == 0) {
+            this->timer4 = 30;
+            Math_Vec3f_Copy(&sp2C, &this->actor.posRot.pos);
+            sp2C.y += this->actor.waterY;
+            func_80029444(globalCtx, &sp2C, 100, 500, 30);
+        }
+
+        if (this->actor.bgCheckFlags & 8) {
+            this->actor.velocity.y = 10.0f;
+            this->actor.speedXZ = 1.0f;
+        }
+    } else {
+        this->actor.gravity = -2.0f;
+
+        if (this->actor.bgCheckFlags & 8) {
+            this->actor.velocity.y = 10.0f;
+            this->actor.speedXZ = 1.0f;
+            this->actor.gravity = 0.0f;
+        } else {
+            this->actor.speedXZ = 4.0f;
+        }
+
+        if (this->actor.bgCheckFlags & 1) {
+            this->actor.gravity = -2.0f;
+            this->timer6 = 100;
+            this->timer4 = 0;
+            this->actor.velocity.y = 0.0f;
+
+            if (this->unk_2A8 == 0) {
+                this->actionFunc = func_80AB7290;
+            } else {
+                this->actionFunc = func_80AB7204;
+            }
+        }
+    }
+
+    func_80AB5BF8(this, globalCtx, 2);
+}
+
+void func_80AB70A0(EnNiw* this, GlobalContext* globalCtx) {
+    func_800800F8(globalCtx, 0x8F2, -0x63, &this->actor, 0);
+    this->timer5 = 100;
+    this->unk_2A2 = 1;
+    this->actionFunc = func_80AB70F8;
+}
+
+void func_80AB70F8(EnNiw* this, GlobalContext* globalCtx) {
+    this->sfxTimer1 = 100;
+
+    if (this->timer5 == 0) {
+        this->timer5 = 60;
+        this->timer1 = 10;
+        this->unk_2A2 = 4;
+        this->actionFunc = func_80AB714C;
+    }
+
+    func_80AB5BF8(this, globalCtx, this->unk_2A2);
+}
+
+void func_80AB714C(EnNiw* this, GlobalContext* globalCtx) {
+    this->sfxTimer1 = 100;
+
+    if (this->timer5 == 40) {
+        this->unk_26C[0] = 10000.0f;
+        this->unk_26C[7] = 14000.0f;
+        this->unk_26C[5] = 14000.0f;
+        this->unk_26C[6] = 0.0f;
+        this->unk_26C[8] = 0.0f;
+        this->unk_26C[1] = 0.0f;
+        this->unk_26C[2] = 0.0f;
+        this->timer1 = 10;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M);
+    }
+
+    if (this->timer5 == 0) {
+        this->timer7 = 10;
+        this->unk_2E4 = this->actor.yawTowardsLink;
+        this->actor.flags &= ~1;
+        this->actionFunc = func_80AB7204;
+    }
+
+    func_80AB5BF8(&this->actor, globalCtx, this->unk_2A2);
+}
+
+void func_80AB7204(EnNiw* this, GlobalContext* globalCtx) {
+    EnNiw_SpawnAttackCucco(this, globalCtx);
+    if (this->timer7 < 2) {
+        if (this->timer7 == 1) {
+            this->actor.speedXZ = 3.0f;
+            this->unk_2A0 = Math_Rand_ZeroFloat(1.99f);
+            this->timer1 = this->timer2 = this->timer3 = this->timer4 = 0;
+            return;
+        }
+        func_80AB6100(this, globalCtx, 1);
```

Comment:
I would use an `else` here instead of an early return since there's no action setup or actor_kill